rclpy/rclpy/action/client.py

```diff
@@ -120,11 +158,18 @@ def __init__(
         )
 
         self._is_ready = False
+
+        # key: UUID in bytes, value: ClientGoalHandle
```

Comment:
Recommend `value` being a [weak reference](https://docs.python.org/3/library/weakref.html) to a void a reference cycle

Reply:
Now weak references: e2df735